specs/01-sharded-archiving.md

```diff
@@ -0,0 +1,288 @@
+- New pieces need to be stored by farmers and made available for retrieval by other shards and nodes
+  in the system.
+- Pieces from a shard are uniquely identified through an sequentially increasing `piece_index`.
+- Output: Segment with `NUM_PIECES` pieces and a `segment_root`.
+
+#### 7. Segment proof Submission to Beacon Chain
+
+- Input: New segments from a shard.
+- As soon as a new block is produced that includes the creation of a new segment, the segment proof
+  is submitted as a transaction in the beacon chain to commit the segment into the global history of
+  the system.
+- To submit this information to the beacon chain a `SegmentInfo` transaction is created that
+  includes the `segment_root`, the `segment_index`, and the `shard_id` of the shard where the
+  segment was created.
+- Any node in the shard (from farmers to full-nodes) are allowed to create the `SegmentInfo`
+  transaction that submits the segments into the parent's shard history.
+
+```rust
+type SegmentInfo = {
+  segment_root: Vec<u8>, // 32B
+  segment_index: u64, // 8B
+  shard_id: ShardId, // 4B
+}
+```
+
+- Output: `SegmentInfo` transaction in the beacon chain mempool.
+
+#### 8. Segment proof in beacon chain and Super Segment Construction
+
+- Input: Transaction with `SegmentInfo` in the beacon chain.
+- The execution of the `SegmentInfo` transaction when included in a block of the beacon chain
+  triggers the proof of the shard's segment in the global history of the beacon chain.
+- To verify the `SegmentInfo` and execute the transaction, nodes verify the following:
+  - Ensure `segment_index` is the subsequent index after the last one committed for the shard.
+- If the verification of the `SegmentInfo` is successful, the child shard segment is added to the
+  beacon chain shard's history by adding the shard's `segment_root` to the `segment_roots[]` that
+  indexes the all proofs from the global history (including beacon chain and all child segments).
+- Apart from adding the segment proof in the global history, a new `SuperSegment` is created
+  aggregating all the segment proofs included in a block of the beacon chain. The goal for this
+  super segment proof is to simplify the verification of segments by light clients.
+- The `super_segment` is created by aggregating the segment proofs of the child segments and
+  creating a new proof that is included in the parent shard's history committing (and wrapping) all
+  the history of shards in the lower layers of the architecture. This `super_segment` proof is
+  directly a Merkle proof of all of the shard segments and child super segment proofs from that
+  shard: `C_super_segment = Merkle_Proof(S1, ..., Sn, SS1, ..., SSn)`.
+- Along with `C_super_segment` the corresponding `ws1` segments proof are created.
+- When a new super segment is created instead of broadcasting the network the full super segment,
+  only the `SuperSegmentHeader` is broadcasted, allowing nodes to proactively determine if they
```

Comment:
```suggestion
- When a new super segment is created instead of broadcasting to the network the full super segment,
  only the `SuperSegmentHeader` is broadcast, allowing nodes to proactively determine if they
```

I'm not sure how can nodes proactively determine just from a Merkle Root what commitments they have and what commitments they don't. In fact this `SuperSegmentHeadder` is never broadcast, it is composed locally by the block producer using segments they see (if they see any at all) in the transaction pool.